google/resumable_media/requests/_helpers.py

```diff
@@ -88,11 +88,14 @@ def http_request(transport, method, url, data=None, headers=None,
             may also add additional headers).
         retry_strategy (~google.resumable_media.common.RetryStrategy): The
             strategy to use if the request fails and must be retried.
+        transport_kwargs (Dict[str, str]): Extra keyword arguments to be
+            passed along to ``transport.request``.
 
     Returns:
         ~requests.Response: The return value of ``transport.request()``.
     """
     func = functools.partial(
-        transport.request, method, url, data=data, headers=headers)
+        transport.request, method, url, data=data, headers=headers,
+        **transport_kwargs)
```

Comment:
@jonparrott I just realized that allowing `stream=True` to go through here isn't a free lunch, because `_helpers.wait_and_retry()` expects to be retrying a failure for a single request, but ISTM the socket might fail midway through?

Any ideas how we could mock and test this with `urllib3` / `requests`?